File: lib/data/reagent.dart
import 'package:flutter/material.dart';

class Reagent {
  const Reagent({
    required this.id,
    required this.displayNameEn,
    required this.displayNameAr,
    required this.dropperTint,
  });

  final String id;
  final String displayNameEn;
  final String displayNameAr;
  final Color dropperTint;
}

File: Assets/_Project/Scripts/Data/Reagent.cs
using UnityEngine;

namespace Kemika.Data
{
    [CreateAssetMenu(fileName = "NewReagent", menuName = "Kemika/Reagent")]
    public class Reagent : ScriptableObject
    {
        public string id;
        public string displayNameEn;
        public string displayNameAr;
        public Color dropperTint = Color.white;

        public static Reagent Create(string id, string nameEn, string nameAr, Color tint)
        {
            var reagent = CreateInstance<Reagent>();
            reagent.id = id;
            reagent.displayNameEn = nameEn;
            reagent.displayNameAr = nameAr;
            reagent.dropperTint = tint;
            return reagent;
        }
    }
}
